RESTResource/RESTResource.js

```diff
@@ -256,6 +256,38 @@ export function substitute(original, props, state, module, logger, dataKey) {
   return result;
 }
 
+/**
+ * buildOption
+ * Some manifest options properties may be an object or a function, determine which
+ * the passed property is and act accordingly
+ *
+ * @param option object or function
+ * @param props object
+ * @param state object
+ * @param module string name of the module the resource is affiliated with
+ * @param logger object a logger
+ * @param dataKey string unique key to disambiguate this resource from another
+ * which otherwise has the same attributes
+ *
+ * @return object
+ */
+
+export function buildOption(option, props, state, module, logger, dataKey) {
+  let toReturn;
+  if (typeof option === 'object') {
+    toReturn = _.mapValues(
+      option,
+      param => substitute(param, props, state, module, logger, dataKey)
+    );
+    const someParamIsNull = Object.values(option).some(value => value === null);
+    if (someParamIsNull) return null; // or toReturn = null
+  } else if (typeof option === 'function') {
+    const parsedQuery = queryString.parse(props.location?.search);
+    toReturn = option(parsedQuery, props.match?.params, mockProps(state, module, dataKey, logger).resources, logger, props);
+  }
+  return toReturn;
+}
```

Comment:
I would avoid if else, as early return can be used. Also `someParamIsNull` check can be made before computing the option.
```suggestion
export function buildOption(option, props, state, module, logger, dataKey) {
  if (typeof option === 'object') {
    const someParamIsNull = Object.values(option).some(value => value === null);
    
    if (someParamIsNull) return null;

    return _.mapValues(
      option,
      param => substitute(param, props, state, module, logger, dataKey)
    );
  }

   if (typeof option === 'function') {
    const parsedQuery = queryString.parse(props.location?.search);
    return option(parsedQuery, props.match?.params, mockProps(state, module, dataKey, logger).resources, logger, props);
  }

  return null;
}
```

Reply:
Thanks for the suggestion @ViktorSoroka07. I think this is pretty subjective to be honest. In my mind the if/else is more explicit than the early returns. I think your suggestion is perfectly valid, I just don't really know if it gives us anything, other than a different approach :)

Reply:
So for me is early return is better Andrew, but this is really the style of preference:). So the only useful moment here still 
```
    toReturn = _.mapValues(
      option,
      param => substitute(param, props, state, module, logger, dataKey)
    );
    const someParamIsNull = Object.values(option).some(value => value === null);
    if (someParamIsNull) return null; // or toReturn = null
```
to move the `if (someParamIsNull) return null; // or toReturn = null` check before the `toReturn`.

Reply:
Thanks for that @ViktorSoroka07 I'm very grateful for your feedback. I've now moved the check for null values and the subsequent return to above the formation of `toReturn`, you're right, it removes the unnecessary formation of `toReturn`.